{
  "comment": [
    "AUTOGENERATED FILE",
    "DO NOT EDIT MANUALLY",
    "The purpose of this file is to show the combined model, after all the links",
    "between models are resolved and the final model is constructed."
  ],
  "model": {
    "types": {
      "API": {
        "name": "API",
        "meta": {
          "description": "Errors originating in the web3 API."
        },
        "bindings": {
          "rust": {
            "name": "Box<API>",
            "path": ""
          },
          "typescript": {
            "name": "Box<API>",
            "path": ""
          }
        }
      },
      "AnvilEnvironment": {
        "name": "AnvilEnvironment",
        "meta": {
          "description": ""
        },
        "bindings": {
          "rust": {
            "name": "Box<AnvilEnvironment>",
            "path": ""
          },
          "typescript": {
            "name": "Box<AnvilEnvironment>",
            "path": ""
          }
        }
      },
      "EraVM": {
        "name": "EraVM",
        "meta": {
          "description": "Errors in EraVM virtual machine executing contracts."
        },
        "bindings": {
          "rust": {
            "name": "Box<EraVM>",
            "path": ""
          },
          "typescript": {
            "name": "Box<EraVM>",
            "path": ""
          }
        }
      },
      "ExecutionPlatform": {
        "name": "ExecutionPlatform",
        "meta": {
          "description": "Errors in the contract execution environment, bootloader, etc."
        },
        "bindings": {
          "rust": {
            "name": "Box<ExecutionPlatform>",
            "path": ""
          },
          "typescript": {
            "name": "Box<ExecutionPlatform>",
            "path": ""
          }
        }
      },
      "FoundryUpstream": {
        "name": "FoundryUpstream",
        "meta": {
          "description": "Errors originating in the upstream Foundry implementation."
        },
        "bindings": {
          "rust": {
            "name": "Box<FoundryUpstream>",
            "path": ""
          },
          "typescript": {
            "name": "Box<FoundryUpstream>",
            "path": ""
          }
        }
      },
      "FoundryZksync": {
        "name": "FoundryZksync",
        "meta": {
          "description": "Errors originating in the ZKsync codebase for Foundry."
        },
        "bindings": {
          "rust": {
            "name": "Box<FoundryZksync>",
            "path": ""
          },
          "typescript": {
            "name": "Box<FoundryZksync>",
            "path": ""
          }
        }
      },
      "Halt": {
        "name": "Halt",
        "meta": {
          "description": ""
        },
        "bindings": {
          "rust": {
            "name": "Box<Halt>",
            "path": ""
          },
          "typescript": {
            "name": "Box<Halt>",
            "path": ""
          }
        }
      },
      "HardhatUpstream": {
        "name": "HardhatUpstream",
        "meta": {
          "description": "Errors originating in the upstream Hardhat implementation."
        },
        "bindings": {
          "rust": {
            "name": "Box<HardhatUpstream>",
            "path": ""
          },
          "typescript": {
            "name": "Box<HardhatUpstream>",
            "path": ""
          }
        }
      },
      "HardhatZksync": {
        "name": "HardhatZksync",
        "meta": {
          "description": "Errors originating in the ZKsync codebase for HardHat."
        },
        "bindings": {
          "rust": {
            "name": "Box<HardhatZksync>",
            "path": ""
          },
          "typescript": {
            "name": "Box<HardhatZksync>",
            "path": ""
          }
        }
      },
      "LLVM_EVM": {
        "name": "LLVM_EVM",
        "meta": {
          "description": "Errors originating in the compiler backend for Ethereum VM (EVM)."
        },
        "bindings": {
          "rust": {
            "name": "Box<LLVM_EVM>",
            "path": ""
          },
          "typescript": {
            "name": "Box<LLVM_EVM>",
            "path": ""
          }
        }
      },
      "LLVM_Era": {
        "name": "LLVM_Era",
        "meta": {
          "description": "Errors originating in the compiler backend for EraVM."
        },
        "bindings": {
          "rust": {
            "name": "Box<LLVM_Era>",
            "path": ""
          },
          "typescript": {
            "name": "Box<LLVM_Era>",
            "path": ""
          }
        }
      },
      "Revert": {
        "name": "Revert",
        "meta": {
          "description": ""
        },
        "bindings": {
          "rust": {
            "name": "Box<Revert>",
            "path": ""
          },
          "typescript": {
            "name": "Box<Revert>",
            "path": ""
          }
        }
      },
      "Sequencer": {
        "name": "Sequencer",
        "meta": {
          "description": "Errors in the sequencer node"
        },
        "bindings": {
          "rust": {
            "name": "Box<Sequencer>",
            "path": ""
          },
          "typescript": {
            "name": "Box<Sequencer>",
            "path": ""
          }
        }
      },
      "Solc": {
        "name": "Solc",
        "meta": {
          "description": "Errors originating in the official Solidity compiler."
        },
        "bindings": {
          "rust": {
            "name": "Box<Solc>",
            "path": ""
          },
          "typescript": {
            "name": "Box<Solc>",
            "path": ""
          }
        }
      },
      "SolcFork": {
        "name": "SolcFork",
        "meta": {
          "description": "Errors originating in the ZKsync fork of Solidity compiler."
        },
        "bindings": {
          "rust": {
            "name": "Box<SolcFork>",
            "path": ""
          },
          "typescript": {
            "name": "Box<SolcFork>",
            "path": ""
          }
        }
      },
      "StateLoader": {
        "name": "StateLoader",
        "meta": {
          "description": ""
        },
        "bindings": {
          "rust": {
            "name": "Box<StateLoader>",
            "path": ""
          },
          "typescript": {
            "name": "Box<StateLoader>",
            "path": ""
          }
        }
      },
      "WrappedError": {
        "name": "WrappedError",
        "meta": {
          "description": "Any wrapped error"
        },
        "bindings": {
          "rust": {
            "name": "serde_json::Value",
            "path": ""
          }
        }
      },
      "Zksolc": {
        "name": "Zksolc",
        "meta": {
          "description": "Errors originating in the ZKsync Solidity compiler for EraVM and EVM."
        },
        "bindings": {
          "rust": {
            "name": "Box<Zksolc>",
            "path": ""
          },
          "typescript": {
            "name": "Box<Zksolc>",
            "path": ""
          }
        }
      },
      "Zkvyper": {
        "name": "Zkvyper",
        "meta": {
          "description": "Errors originating in the ZKsync Vyper compiler for EraVM."
        },
        "bindings": {
          "rust": {
            "name": "Box<Zkvyper>",
            "path": ""
          },
          "typescript": {
            "name": "Box<Zkvyper>",
            "path": ""
          }
        }
      },
      "bytes": {
        "name": "bytes",
        "meta": {
          "description": "Sequence of bytes"
        },
        "bindings": {
          "rust": {
            "name": "Vec<u8>",
            "path": ""
          }
        }
      },
      "int": {
        "name": "int",
        "meta": {
          "description": "Signed 32-bit integer"
        },
        "bindings": {
          "rust": {
            "name": "i32",
            "path": ""
          }
        }
      },
      "map": {
        "name": "map",
        "meta": {
          "description": "String to string mapping."
        },
        "bindings": {
          "rust": {
            "name": "Vec<(String,String)>",
            "path": ""
          }
        }
      },
      "string": {
        "name": "string",
        "meta": {
          "description": "Unicode zero-terminated string"
        },
        "bindings": {
          "rust": {
            "name": "String",
            "path": ""
          }
        }
      },
      "uint": {
        "name": "uint",
        "meta": {
          "description": "Unsigned 32-bit integer"
        },
        "bindings": {
          "rust": {
            "name": "u32",
            "path": ""
          }
        }
      }
    },
    "domains": {
      "Anvil": {
        "name": "Anvil",
        "code": 5,
        "components": [
          "AnvilEnvironment",
          "Halt",
          "Revert",
          "StateLoader"
        ],
        "bindings": {
          "rust": "Anvil",
          "typescript": "Anvil"
        },
        "identifier": "anvil",
        "description": "Errors originating in Anvil for ZKsync."
      },
      "Compiler": {
        "name": "Compiler",
        "code": 2,
        "components": [
          "LLVM_EVM",
          "LLVM_Era",
          "Solc",
          "SolcFork",
          "Zksolc",
          "Zkvyper"
        ],
        "bindings": {
          "rust": "Compiler",
          "typescript": "Compiler"
        },
        "identifier": "compiler",
        "description": "Compilers such as zksolc, zkvyper, solc."
      },
      "Core": {
        "name": "Core",
        "code": 1,
        "components": [
          "API",
          "EraVM",
          "ExecutionPlatform",
          "Sequencer"
        ],
        "bindings": {
          "rust": "Core",
          "typescript": "Core"
        },
        "identifier": "core",
        "description": "Errors in core ZKsync components such as sequencer or mempool."
      },
      "Foundry": {
        "name": "Foundry",
        "code": 4,
        "components": [
          "FoundryUpstream",
          "FoundryZksync"
        ],
        "bindings": {
          "rust": "Foundry",
          "typescript": "Foundry"
        },
        "identifier": "foundry",
        "description": "Errors originating in Foundry."
      },
      "Hardhat": {
        "name": "Hardhat",
        "code": 3,
        "components": [
          "HardhatUpstream",
          "HardhatZksync"
        ],
        "bindings": {
          "rust": "Hardhat",
          "typescript": "Hardhat"
        },
        "identifier": "hardhat",
        "description": "Errors originating in HardHat."
      }
    },
    "components": {
      "API": {
        "name": "API",
        "code": 4,
        "domain_name": "Core",
        "bindings": {
          "rust": "API",
          "typescript": "API"
        },
        "identifier": "api",
        "description": "Errors originating in the web3 API."
      },
      "AnvilEnvironment": {
        "name": "AnvilEnvironment",
        "code": 1,
        "domain_name": "Anvil",
        "bindings": {
          "rust": "AnvilEnvironment",
          "typescript": "AnvilEnvironment"
        },
        "identifier": "env",
        "description": ""
      },
      "EraVM": {
        "name": "EraVM",
        "code": 2,
        "domain_name": "Core",
        "bindings": {
          "rust": "EraVM",
          "typescript": "EraVM"
        },
        "identifier": "eravm",
        "description": "Errors in EraVM virtual machine executing contracts."
      },
      "ExecutionPlatform": {
        "name": "ExecutionPlatform",
        "code": 3,
        "domain_name": "Core",
        "bindings": {
          "rust": "ExecutionPlatform",
          "typescript": "ExecutionPlatform"
        },
        "identifier": "exec",
        "description": "Errors in the contract execution environment, bootloader, etc."
      },
      "FoundryUpstream": {
        "name": "FoundryUpstream",
        "code": 1,
        "domain_name": "Foundry",
        "bindings": {
          "rust": "FoundryUpstream",
          "typescript": "FoundryUpstream"
        },
        "identifier": "upstream",
        "description": "Errors originating in the upstream Foundry implementation."
      },
      "FoundryZksync": {
        "name": "FoundryZksync",
        "code": 2,
        "domain_name": "Foundry",
        "bindings": {
          "rust": "FoundryZksync",
          "typescript": "FoundryZksync"
        },
        "identifier": "zksync",
        "description": "Errors originating in the ZKsync codebase for Foundry."
      },
      "Halt": {
        "name": "Halt",
        "code": 3,
        "domain_name": "Anvil",
        "bindings": {
          "rust": "Halt",
          "typescript": "Halt"
        },
        "identifier": "halt",
        "description": ""
      },
      "HardhatUpstream": {
        "name": "HardhatUpstream",
        "code": 1,
        "domain_name": "Hardhat",
        "bindings": {
          "rust": "HardhatUpstream",
          "typescript": "HardhatUpstream"
        },
        "identifier": "upstream",
        "description": "Errors originating in the upstream Hardhat implementation."
      },
      "HardhatZksync": {
        "name": "HardhatZksync",
        "code": 2,
        "domain_name": "Hardhat",
        "bindings": {
          "rust": "HardhatZksync",
          "typescript": "HardhatZksync"
        },
        "identifier": "zksync",
        "description": "Errors originating in the ZKsync codebase for HardHat."
      },
      "LLVM_EVM": {
        "name": "LLVM_EVM",
        "code": 2,
        "domain_name": "Compiler",
        "bindings": {
          "rust": "LLVM_EVM",
          "typescript": "LLVM_EVM"
        },
        "identifier": "llvm+evm",
        "description": "Errors originating in the compiler backend for Ethereum VM (EVM)."
      },
      "LLVM_Era": {
        "name": "LLVM_Era",
        "code": 1,
        "domain_name": "Compiler",
        "bindings": {
          "rust": "LLVM_Era",
          "typescript": "LLVM_Era"
        },
        "identifier": "llvm+era",
        "description": "Errors originating in the compiler backend for EraVM."
      },
      "Revert": {
        "name": "Revert",
        "code": 4,
        "domain_name": "Anvil",
        "bindings": {
          "rust": "Revert",
          "typescript": "Revert"
        },
        "identifier": "revert",
        "description": ""
      },
      "Sequencer": {
        "name": "Sequencer",
        "code": 1,
        "domain_name": "Core",
        "bindings": {
          "rust": "Sequencer",
          "typescript": "Sequencer"
        },
        "identifier": "seq",
        "description": "Errors in the sequencer node"
      },
      "Solc": {
        "name": "Solc",
        "code": 3,
        "domain_name": "Compiler",
        "bindings": {
          "rust": "Solc",
          "typescript": "Solc"
        },
        "identifier": "solc",
        "description": "Errors originating in the official Solidity compiler."
      },
      "SolcFork": {
        "name": "SolcFork",
        "code": 4,
        "domain_name": "Compiler",
        "bindings": {
          "rust": "SolcFork",
          "typescript": "SolcFork"
        },
        "identifier": "solc+fork",
        "description": "Errors originating in the ZKsync fork of Solidity compiler."
      },
      "StateLoader": {
        "name": "StateLoader",
        "code": 2,
        "domain_name": "Anvil",
        "bindings": {
          "rust": "StateLoader",
          "typescript": "StateLoader"
        },
        "identifier": "state",
        "description": ""
      },
      "Zksolc": {
        "name": "Zksolc",
        "code": 5,
        "domain_name": "Compiler",
        "bindings": {
          "rust": "Zksolc",
          "typescript": "Zksolc"
        },
        "identifier": "zksolc",
        "description": "Errors originating in the ZKsync Solidity compiler for EraVM and EVM."
      },
      "Zkvyper": {
        "name": "Zkvyper",
        "code": 6,
        "domain_name": "Compiler",
        "bindings": {
          "rust": "Zkvyper",
          "typescript": "Zkvyper"
        },
        "identifier": "zkvyper",
        "description": "Errors originating in the ZKsync Vyper compiler for EraVM."
      }
    },
    "errors": {
      "[anvil-env-0]": {
        "domain": "Anvil",
        "component": "AnvilEnvironment",
        "name": "GenericError",
        "code": 0,
        "identifier": "[anvil-env-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[anvil-env-1]": {
        "domain": "Anvil",
        "component": "AnvilEnvironment",
        "name": "LogFileAccessError",
        "code": 1,
        "identifier": "[anvil-env-1]",
        "message": "Unable to access log file: {log_filename}",
        "fields": [
          {
            "name": "log_filename",
            "type": "string"
          },
          {
            "name": "wrapped_error",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "Anvil was unable to open the log file for writing.",
          "summary": "Unable to access log file.",
          "likely_causes": [
            {
              "cause": "Not enough space.",
              "fixes": [
                "Check if you have enough free space in your storage to create or open a log file"
              ],
              "report": "",
              "owner": null,
              "references": []
            },
            {
              "cause": "Insufficient permissions.",
              "fixes": [
                "Check if you have permissions to write to the log file."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "LogFileAccessError",
            "path": ""
          },
          "typescript": {
            "name": "LogFileAccessError",
            "path": ""
          }
        }
      },
      "[anvil-env-2]": {
        "domain": "Anvil",
        "component": "AnvilEnvironment",
        "name": "UnsupportedMethodError",
        "code": 2,
        "identifier": "[anvil-env-2]",
        "message": "Unsupported method.",
        "fields": [],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "UnsupportedMethodError",
            "path": ""
          },
          "typescript": {
            "name": "UnsupportedMethodError",
            "path": ""
          }
        }
      },
      "[anvil-env-3]": {
        "domain": "Anvil",
        "component": "AnvilEnvironment",
        "name": "Web3Error",
        "code": 3,
        "identifier": "[anvil-env-3]",
        "message": "Web3Error: {inner}.",
        "fields": [
          {
            "name": "inner",
            "type": "API"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "Web3Error",
            "path": ""
          },
          "typescript": {
            "name": "Web3Error",
            "path": ""
          }
        }
      },
      "[anvil-halt-0]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "GenericError",
        "code": 0,
        "identifier": "[anvil-halt-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[anvil-halt-10]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "UnexpectedVMBehavior",
        "code": 10,
        "identifier": "[anvil-halt-10]",
        "message": "virtual machine entered unexpected state. Please contact developers and provide transaction details that caused this error. Error description: {problem}",
        "fields": [
          {
            "name": "problem",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "This error is used temporarily instead of panics to provide a better experience for developers. It indicates that the VM entered an unforeseen state during transaction execution.",
          "summary": "The virtual machine encountered an unexpected state.",
          "likely_causes": [
            {
              "cause": "Internal VM logic error or unforeseen edge case.",
              "fixes": [
                "Contact the development team with detailed transaction information.",
                "Review the transaction logic for potential issues."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "UnexpectedVMBehavior",
            "path": ""
          },
          "typescript": {
            "name": "UnexpectedVMBehavior",
            "path": ""
          }
        }
      },
      "[anvil-halt-11]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "BootloaderOutOfGas",
        "code": 11,
        "identifier": "[anvil-halt-11]",
        "message": "Bootloader out of gas",
        "fields": [],
        "documentation": {
          "description": "This error occurs when the bootloader does not have enough gas to continue executing the transaction.",
          "summary": "The bootloader has run out of gas.",
          "likely_causes": [
            {
              "cause": "Insufficient gas limit set for the transaction.",
              "fixes": [
                "Increase the gas limit for the transaction.",
                "Optimize the transaction to consume less gas."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "BootloaderOutOfGas",
            "path": ""
          },
          "typescript": {
            "name": "BootloaderOutOfGas",
            "path": ""
          }
        }
      },
      "[anvil-halt-12]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "ValidationOutOfGas",
        "code": 12,
        "identifier": "[anvil-halt-12]",
        "message": "Validation run out of gas",
        "fields": [],
        "documentation": {
          "description": "This error is emitted when the validation phase of transaction execution exceeds the allocated gas limit.",
          "summary": "The validation step ran out of gas.",
          "likely_causes": [
            {
              "cause": "Complex validation logic requiring more gas than allocated.",
              "fixes": [
                "Increase the gas limit for the validation step.",
                "Simplify the validation logic to consume less gas."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "ValidationOutOfGas",
            "path": ""
          },
          "typescript": {
            "name": "ValidationOutOfGas",
            "path": ""
          }
        }
      },
      "[anvil-halt-13]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "TooBigGasLimit",
        "code": 13,
        "identifier": "[anvil-halt-13]",
        "message": "Transaction has a too big ergs limit and will not be executed by the server",
        "fields": [],
        "documentation": {
          "description": "This error occurs when the gas limit set for the transaction is too large for the server to handle.",
          "summary": "The transaction's gas limit is excessively high.",
          "likely_causes": [
            {
              "cause": "Setting an unrealistically high gas limit for the transaction.",
              "fixes": [
                "Adjust the gas limit to a reasonable value.",
                "Consult the network's gas limit guidelines to set an appropriate limit."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "TooBigGasLimit",
            "path": ""
          },
          "typescript": {
            "name": "TooBigGasLimit",
            "path": ""
          }
        }
      },
      "[anvil-halt-14]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "NotEnoughGasProvided",
        "code": 14,
        "identifier": "[anvil-halt-14]",
        "message": "Bootloader did not have enough gas to start the transaction",
        "fields": [],
        "documentation": {
          "description": "This error is emitted when the bootloader lacks the necessary gas to begin executing the transaction.",
          "summary": "Insufficient gas provided to initiate the transaction.",
          "likely_causes": [
            {
              "cause": "Insufficient gas allocated for the transaction initiation.",
              "fixes": [
                "Increase the gas provided for the transaction.",
                "Ensure that the gas estimation is accurate before setting the gas limit."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "NotEnoughGasProvided",
            "path": ""
          },
          "typescript": {
            "name": "NotEnoughGasProvided",
            "path": ""
          }
        }
      },
      "[anvil-halt-15]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "MissingInvocationLimitReached",
        "code": 15,
        "identifier": "[anvil-halt-15]",
        "message": "Tx produced too much cold storage accesses",
        "fields": [],
        "documentation": {
          "description": "This error occurs when the transaction makes too many missing invocations to memory, surpassing the allowed limit.",
          "summary": "The transaction exceeded the allowed number of cold storage invocations.",
          "likely_causes": [
            {
              "cause": "Highly complex transaction with excessive memory accesses.",
              "fixes": [
                "Optimize the transaction to reduce the number of memory accesses.",
                "Break down the transaction into smaller, more manageable parts."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "MissingInvocationLimitReached",
            "path": ""
          },
          "typescript": {
            "name": "MissingInvocationLimitReached",
            "path": ""
          }
        }
      },
      "[anvil-halt-16]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "FailedToSetL2Block",
        "code": 16,
        "identifier": "[anvil-halt-16]",
        "message": "Failed to set information about the L2 block: {msg}",
        "fields": [
          {
            "name": "msg",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "This error is emitted when the system fails to set the necessary information for the L2 block during transaction execution.",
          "summary": "Unable to set L2 block information.",
          "likely_causes": [
            {
              "cause": "Invalid L2 block data or internal processing error.",
              "fixes": [
                "Verify the L2 block information being set.",
                "Check for any internal processing issues that might prevent setting the L2 block."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "FailedToSetL2Block",
            "path": ""
          },
          "typescript": {
            "name": "FailedToSetL2Block",
            "path": ""
          }
        }
      },
      "[anvil-halt-17]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "FailedToAppendTransactionToL2Block",
        "code": 17,
        "identifier": "[anvil-halt-17]",
        "message": "Failed to append the transaction to the current L2 block: {msg}",
        "fields": [
          {
            "name": "msg",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "This error occurs when the system cannot append the transaction to the ongoing L2 block, possibly due to data inconsistencies or internal errors.",
          "summary": "Failed to add the transaction to the L2 block.",
          "likely_causes": [
            {
              "cause": "Data inconsistency or corruption in the L2 block.",
              "fixes": [
                "Ensure the integrity of the L2 block data.",
                "Retry appending the transaction after verifying the L2 block state."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "FailedToAppendTransactionToL2Block",
            "path": ""
          },
          "typescript": {
            "name": "FailedToAppendTransactionToL2Block",
            "path": ""
          }
        }
      },
      "[anvil-halt-18]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "VMPanic",
        "code": 18,
        "identifier": "[anvil-halt-18]",
        "message": "VM panicked",
        "fields": [],
        "documentation": {
          "description": "This error is emitted when the VM experiences a critical failure and panics during transaction execution.",
          "summary": "The virtual machine encountered a panic.",
          "likely_causes": [
            {
              "cause": "Uncaught exception or critical internal error within the VM.",
              "fixes": [
                "Review the transaction for any operations that might cause the VM to panic.",
                "Consult the development team to investigate the VM's internal state."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "VMPanic",
            "path": ""
          },
          "typescript": {
            "name": "VMPanic",
            "path": ""
          }
        }
      },
      "[anvil-halt-19]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "TracerCustom",
        "code": 19,
        "identifier": "[anvil-halt-19]",
        "message": "Tracer aborted execution: {msg}",
        "fields": [
          {
            "name": "msg",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "This error occurs when a custom tracer used during transaction execution decides to abort the process, typically due to specific conditions being met.",
          "summary": "Tracer aborted the transaction execution.",
          "likely_causes": [
            {
              "cause": "Custom tracer conditions not met or detected an issue.",
              "fixes": [
                "Review the tracer's logic to understand why it aborted the execution.",
                "Adjust the tracer conditions if necessary to prevent unnecessary aborts."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "TracerCustom",
            "path": ""
          },
          "typescript": {
            "name": "TracerCustom",
            "path": ""
          }
        }
      },
      "[anvil-halt-1]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "ValidationFailed",
        "code": 1,
        "identifier": "[anvil-halt-1]",
        "message": "Account validation error: {msg}: {data}",
        "fields": [
          {
            "name": "msg",
            "type": "string"
          },
          {
            "name": "data",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "This error occurs when the account validation step fails during the verification and execution of a transaction.",
          "summary": "Account validation failed during execution.",
          "likely_causes": [
            {
              "cause": "Invalid account state or credentials.",
              "fixes": [
                "Ensure the account has been properly deployed and initialized.",
                "Verify that the account credentials are correct."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "ValidationFailed",
            "path": ""
          },
          "typescript": {
            "name": "ValidationFailed",
            "path": ""
          }
        }
      },
      "[anvil-halt-20]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "FailedToPublishCompressedBytecodes",
        "code": 20,
        "identifier": "[anvil-halt-20]",
        "message": "Failed to publish compressed bytecodes",
        "fields": [],
        "documentation": {
          "description": "This error is emitted when the system fails to publish the compressed bytecodes during transaction execution.",
          "summary": "Unable to publish compressed bytecodes.",
          "likely_causes": [
            {
              "cause": "Compression algorithm failure or network issues during publishing.",
              "fixes": [
                "Ensure the compression algorithm is functioning correctly.",
                "Verify network connectivity and retry the publishing process."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "FailedToPublishCompressedBytecodes",
            "path": ""
          },
          "typescript": {
            "name": "FailedToPublishCompressedBytecodes",
            "path": ""
          }
        }
      },
      "[anvil-halt-21]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "FailedBlockTimestampAssertion",
        "code": 21,
        "identifier": "[anvil-halt-21]",
        "message": "Transaction failed block.timestamp assertion",
        "fields": [],
        "documentation": {
          "description": "This error occurs when the transaction's execution fails to satisfy the `block.timestamp` assertion, indicating a discrepancy in the expected timestamp.",
          "summary": "Block timestamp assertion failed during the transaction.",
          "likely_causes": [
            {
              "cause": "Incorrect block timestamp or transaction timing issues.",
              "fixes": [
                "Ensure that the block timestamp is set correctly.",
                "Verify the transaction timing and adjust if necessary."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "FailedBlockTimestampAssertion",
            "path": ""
          },
          "typescript": {
            "name": "FailedBlockTimestampAssertion",
            "path": ""
          }
        }
      },
      "[anvil-halt-2]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "PaymasterValidationFailed",
        "code": 2,
        "identifier": "[anvil-halt-2]",
        "message": "Paymaster validation error: {msg}: {data}.",
        "fields": [
          {
            "name": "msg",
            "type": "string"
          },
          {
            "name": "data",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "This error is emitted when the paymaster validation process fails during transaction execution.",
          "summary": "Paymaster validation failed.",
          "likely_causes": [
            {
              "cause": "Invalid paymaster parameters or insufficient funds.",
              "fixes": [
                "Check the paymaster configuration and ensure all parameters are correct.",
                "Ensure the paymaster account has sufficient funds to cover the transaction fees."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "PaymasterValidationFailed",
            "path": ""
          },
          "typescript": {
            "name": "PaymasterValidationFailed",
            "path": ""
          }
        }
      },
      "[anvil-halt-3]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "PrePaymasterPreparationFailed",
        "code": 3,
        "identifier": "[anvil-halt-3]",
        "message": "Pre-paymaster preparation error: {msg}: {data}",
        "fields": [
          {
            "name": "msg",
            "type": "string"
          },
          {
            "name": "data",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "This error occurs when the system fails to prepare the paymaster before executing a transaction.",
          "summary": "Pre-paymaster preparation step failed.",
          "likely_causes": [
            {
              "cause": "Error in setting up pre-paymaster state.",
              "fixes": [
                "Verify the pre-paymaster setup process and ensure all prerequisites are met.",
                "Check for any dependencies or configurations that might be missing."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "PrePaymasterPreparationFailed",
            "path": ""
          },
          "typescript": {
            "name": "PrePaymasterPreparationFailed",
            "path": ""
          }
        }
      },
      "[anvil-halt-4]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "PayForTxFailed",
        "code": 4,
        "identifier": "[anvil-halt-4]",
        "message": "Failed to pay for the transaction: {msg}: {data}",
        "fields": [
          {
            "name": "msg",
            "type": "string"
          },
          {
            "name": "data",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "This error is emitted when the system fails to deduct the required fees for executing the transaction.",
          "summary": "Payment for the transaction failed.",
          "likely_causes": [
            {
              "cause": "Insufficient balance or incorrect payment parameters.",
              "fixes": [
                "Ensure the sender has enough balance to cover the transaction fees.",
                "Verify that the payment parameters are correctly specified."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "PayForTxFailed",
            "path": ""
          },
          "typescript": {
            "name": "PayForTxFailed",
            "path": ""
          }
        }
      },
      "[anvil-halt-5]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "FailedToMarkFactoryDependencies",
        "code": 5,
        "identifier": "[anvil-halt-5]",
        "message": "Failed to mark factory dependencies: {msg}: {data}",
        "fields": [
          {
            "name": "msg",
            "type": "string"
          },
          {
            "name": "data",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "This error occurs when the system cannot mark the necessary factory dependencies required for the transaction.",
          "summary": "Failed to mark factory dependencies during execution.",
          "likely_causes": [
            {
              "cause": "Dependency resolution issues or corrupted state.",
              "fixes": [
                "Check the dependency configurations and ensure all required dependencies are available.",
                "Verify the integrity of the system state to rule out corruption."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "FailedToMarkFactoryDependencies",
            "path": ""
          },
          "typescript": {
            "name": "FailedToMarkFactoryDependencies",
            "path": ""
          }
        }
      },
      "[anvil-halt-6]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "FailedToChargeFee",
        "code": 6,
        "identifier": "[anvil-halt-6]",
        "message": "Failed to charge fee: {msg}: {data}",
        "fields": [
          {
            "name": "msg",
            "type": "string"
          },
          {
            "name": "data",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "This error is emitted when the system cannot deduct the necessary fee for the transaction.",
          "summary": "Charging the transaction fee failed.",
          "likely_causes": [
            {
              "cause": "Insufficient funds or incorrect fee parameters.",
              "fixes": [
                "Ensure the account has sufficient funds to cover the transaction fee.",
                "Verify that the fee parameters are correctly specified."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "FailedToChargeFee",
            "path": ""
          },
          "typescript": {
            "name": "FailedToChargeFee",
            "path": ""
          }
        }
      },
      "[anvil-halt-7]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "FromIsNotAnAccount",
        "code": 7,
        "identifier": "[anvil-halt-7]",
        "message": "Sender is not an account",
        "fields": [],
        "documentation": {
          "description": "This error occurs when a transaction is attempted from an address that has not been deployed as an account, meaning the `from` address is just a contract.",
          "summary": "The sender address is not a valid account.",
          "likely_causes": [
            {
              "cause": "Using a contract address as the sender instead of an account.",
              "fixes": [
                "Ensure that the `from` address is a properly deployed account.",
                "Deploy an account if necessary and use its address as the sender."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "FromIsNotAnAccount",
            "path": ""
          },
          "typescript": {
            "name": "FromIsNotAnAccount",
            "path": ""
          }
        }
      },
      "[anvil-halt-8]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "InnerTxError",
        "code": 8,
        "identifier": "[anvil-halt-8]",
        "message": "Bootloader-based tx failed",
        "fields": [],
        "documentation": {
          "description": "This error is currently not used and should be removed when refactoring errors.",
          "summary": "An inner transaction error occurred.",
          "likely_causes": [
            {
              "cause": "Internal transaction processing issue.",
              "fixes": [
                "Await error refactoring to address this issue properly."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "InnerTxError",
            "path": ""
          },
          "typescript": {
            "name": "InnerTxError",
            "path": ""
          }
        }
      },
      "[anvil-halt-9]": {
        "domain": "Anvil",
        "component": "Halt",
        "name": "Unknown",
        "code": 9,
        "identifier": "[anvil-halt-9]",
        "message": "Unknown reason: {msg}: {data}",
        "fields": [
          {
            "name": "msg",
            "type": "string"
          },
          {
            "name": "data",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "This error is emitted when the system encounters an unspecified reason for halting.",
          "summary": "An unknown error occurred.",
          "likely_causes": [
            {
              "cause": "Unexpected or unhandled VM state.",
              "fixes": [
                "Contact the developers with transaction details that caused this error.",
                "Review the transaction to identify any anomalies."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "Unknown",
            "path": ""
          },
          "typescript": {
            "name": "Unknown",
            "path": ""
          }
        }
      },
      "[anvil-revert-0]": {
        "domain": "Anvil",
        "component": "Revert",
        "name": "GenericError",
        "code": 0,
        "identifier": "[anvil-revert-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[anvil-revert-1]": {
        "domain": "Anvil",
        "component": "Revert",
        "name": "General",
        "code": 1,
        "identifier": "[anvil-revert-1]",
        "message": "General VM error: {msg}",
        "fields": [
          {
            "name": "msg",
            "type": "string"
          },
          {
            "name": "data",
            "type": "bytes"
          }
        ],
        "documentation": {
          "description": "This error represents a general VM failure, containing a message and additional data for debugging.",
          "summary": "A general VM error occurred.",
          "likely_causes": [
            {
              "cause": "An unexpected condition occurred within the VM.",
              "fixes": [
                "Review the transaction details and VM logs for more information.",
                "Contact the development team with the error data."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "General",
            "path": ""
          },
          "typescript": {
            "name": "General",
            "path": ""
          }
        }
      },
      "[anvil-revert-2]": {
        "domain": "Anvil",
        "component": "Revert",
        "name": "InnerTxError",
        "code": 2,
        "identifier": "[anvil-revert-2]",
        "message": "Bootloader-based transaction failed.",
        "fields": [],
        "documentation": {
          "description": "This error is emitted when an inner transaction within the VM fails, typically related to bootloader execution.",
          "summary": "An inner transaction error occurred.",
          "likely_causes": [
            {
              "cause": "Failure in executing a nested transaction within the VM.",
              "fixes": [
                "Check the inner transaction details for errors.",
                "Ensure that all prerequisites for the nested transaction are met."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "InnerTxError",
            "path": ""
          },
          "typescript": {
            "name": "InnerTxError",
            "path": ""
          }
        }
      },
      "[anvil-revert-3]": {
        "domain": "Anvil",
        "component": "Revert",
        "name": "VmError",
        "code": 3,
        "identifier": "[anvil-revert-3]",
        "message": "VM Error",
        "fields": [],
        "documentation": {
          "description": "This error indicates a generic failure within the VM, without specific details.",
          "summary": "A generic VM error.",
          "likely_causes": [
            {
              "cause": "A non-specific internal error within the VM.",
              "fixes": [
                "Review the VM logs for more detailed information.",
                "Contact support with the VM state and error context."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "VmError",
            "path": ""
          },
          "typescript": {
            "name": "VmError",
            "path": ""
          }
        }
      },
      "[anvil-revert-4]": {
        "domain": "Anvil",
        "component": "Revert",
        "name": "Unknown",
        "code": 4,
        "identifier": "[anvil-revert-4]",
        "message": "Unknown VM revert reason: function_selector={function_selector}, data={data}",
        "fields": [
          {
            "name": "function_selector",
            "type": "string"
          },
          {
            "name": "data",
            "type": "string"
          }
        ],
        "documentation": {
          "description": "This error is emitted when the VM encounters a revert reason that is not recognized, including the function selector and raw data for debugging.",
          "summary": "An unknown VM revert reason was encountered.",
          "likely_causes": [
            {
              "cause": "Reverting with an unrecognized error type or corrupted data.",
              "fixes": [
                "Verify the integrity of the transaction data.",
                "Ensure that the function selectors used in the contract are correct.",
                "Contact the development team with the raw error data."
              ],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "Unknown",
            "path": ""
          },
          "typescript": {
            "name": "Unknown",
            "path": ""
          }
        }
      },
      "[anvil-state-0]": {
        "domain": "Anvil",
        "component": "StateLoader",
        "name": "GenericError",
        "code": 0,
        "identifier": "[anvil-state-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[anvil-state-1]": {
        "domain": "Anvil",
        "component": "StateLoader",
        "name": "LoadingStateOverExistingStateError",
        "code": 1,
        "identifier": "[anvil-state-1]",
        "message": "Loading state into a node with existing state is not allowed.",
        "fields": [],
        "documentation": {
          "description": "It is not allowed to load a state overriding the existing node state. If you have a use case for that, please create an issue.",
          "summary": "It is not allowed to load a state overriding the existing node state.",
          "likely_causes": []
        },
        "bindings": {
          "rust": {
            "name": "LoadingStateOverExistingStateError",
            "path": ""
          },
          "typescript": {
            "name": "LoadingStateOverExistingStateError",
            "path": ""
          }
        }
      },
      "[anvil-state-2]": {
        "domain": "Anvil",
        "component": "StateLoader",
        "name": "LoadEmptyStateError",
        "code": 2,
        "identifier": "[anvil-state-2]",
        "message": "Loading a state without blocks is not allowed.",
        "fields": [],
        "documentation": {
          "description": "It is not allowed to load a state without any blocks in it.",
          "summary": "Attempt to load a state with no blocks",
          "likely_causes": [
            {
              "cause": "You meant to provide a different state.",
              "fixes": [],
              "report": "",
              "owner": null,
              "references": []
            }
          ]
        },
        "bindings": {
          "rust": {
            "name": "LoadEmptyStateError",
            "path": ""
          },
          "typescript": {
            "name": "LoadEmptyStateError",
            "path": ""
          }
        }
      },
      "[anvil-state-3]": {
        "domain": "Anvil",
        "component": "StateLoader",
        "name": "StateDecompressionError",
        "code": 3,
        "identifier": "[anvil-state-3]",
        "message": "Failed to decompress state: {details}.",
        "fields": [
          {
            "name": "details",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "StateDecompressionError",
            "path": ""
          },
          "typescript": {
            "name": "StateDecompressionError",
            "path": ""
          }
        }
      },
      "[anvil-state-4]": {
        "domain": "Anvil",
        "component": "StateLoader",
        "name": "StateDeserializationError",
        "code": 4,
        "identifier": "[anvil-state-4]",
        "message": "Failed to deserialize state: {details}",
        "fields": [
          {
            "name": "details",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "StateDeserializationError",
            "path": ""
          },
          "typescript": {
            "name": "StateDeserializationError",
            "path": ""
          }
        }
      },
      "[anvil-state-5]": {
        "domain": "Anvil",
        "component": "StateLoader",
        "name": "UnknownStateVersionError",
        "code": 5,
        "identifier": "[anvil-state-5]",
        "message": "Unknown version of the state: {version}.",
        "fields": [
          {
            "name": "version",
            "type": "uint"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "UnknownStateVersionError",
            "path": ""
          },
          "typescript": {
            "name": "UnknownStateVersionError",
            "path": ""
          }
        }
      },
      "[anvil-state-6]": {
        "domain": "Anvil",
        "component": "StateLoader",
        "name": "LoadStateError",
        "code": 6,
        "identifier": "[anvil-state-6]",
        "message": "Error while attempting to access the state located at `{path}`. Reason: {reason}.",
        "fields": [
          {
            "name": "path",
            "type": "string"
          },
          {
            "name": "reason",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "LoadStateError",
            "path": ""
          },
          "typescript": {
            "name": "LoadStateError",
            "path": ""
          }
        }
      },
      "[compiler-llvm+era-0]": {
        "domain": "Compiler",
        "component": "LLVM_Era",
        "name": "GenericError",
        "code": 0,
        "identifier": "[compiler-llvm+era-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[compiler-llvm+evm-0]": {
        "domain": "Compiler",
        "component": "LLVM_EVM",
        "name": "GenericError",
        "code": 0,
        "identifier": "[compiler-llvm+evm-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[compiler-solc+fork-0]": {
        "domain": "Compiler",
        "component": "SolcFork",
        "name": "GenericError",
        "code": 0,
        "identifier": "[compiler-solc+fork-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[compiler-solc-0]": {
        "domain": "Compiler",
        "component": "Solc",
        "name": "GenericError",
        "code": 0,
        "identifier": "[compiler-solc-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[compiler-zksolc-0]": {
        "domain": "Compiler",
        "component": "Zksolc",
        "name": "GenericError",
        "code": 0,
        "identifier": "[compiler-zksolc-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[compiler-zkvyper-0]": {
        "domain": "Compiler",
        "component": "Zkvyper",
        "name": "GenericError",
        "code": 0,
        "identifier": "[compiler-zkvyper-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[core-api-0]": {
        "domain": "Core",
        "component": "API",
        "name": "GenericError",
        "code": 0,
        "identifier": "[core-api-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[core-eravm-0]": {
        "domain": "Core",
        "component": "EraVM",
        "name": "GenericError",
        "code": 0,
        "identifier": "[core-eravm-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[core-exec-0]": {
        "domain": "Core",
        "component": "ExecutionPlatform",
        "name": "GenericError",
        "code": 0,
        "identifier": "[core-exec-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[core-seq-0]": {
        "domain": "Core",
        "component": "Sequencer",
        "name": "GenericError",
        "code": 0,
        "identifier": "[core-seq-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[core-seq-1]": {
        "domain": "Core",
        "component": "Sequencer",
        "name": "GenericSequencerError",
        "code": 1,
        "identifier": "[core-seq-1]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericSequencerError",
            "path": ""
          },
          "typescript": {
            "name": "GenericSequencerError",
            "path": ""
          }
        }
      },
      "[foundry-upstream-0]": {
        "domain": "Foundry",
        "component": "FoundryUpstream",
        "name": "GenericError",
        "code": 0,
        "identifier": "[foundry-upstream-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[foundry-zksync-0]": {
        "domain": "Foundry",
        "component": "FoundryZksync",
        "name": "GenericError",
        "code": 0,
        "identifier": "[foundry-zksync-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[hardhat-upstream-0]": {
        "domain": "Hardhat",
        "component": "HardhatUpstream",
        "name": "GenericError",
        "code": 0,
        "identifier": "[hardhat-upstream-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      },
      "[hardhat-zksync-0]": {
        "domain": "Hardhat",
        "component": "HardhatZksync",
        "name": "GenericError",
        "code": 0,
        "identifier": "[hardhat-zksync-0]",
        "message": "Generic error: {message}",
        "fields": [
          {
            "name": "message",
            "type": "string"
          }
        ],
        "documentation": null,
        "bindings": {
          "rust": {
            "name": "GenericError",
            "path": ""
          },
          "typescript": {
            "name": "GenericError",
            "path": ""
          }
        }
      }
    }
  }
}